### PR TITLE
Add evaluated tree graph to React reconciler stats

### PR DESCRIPTION
**Pull request by trueadm**
Release notes: none

This generates a tree graph of how all the components in the React reconciliation process get handled. It will massively help debugging what did or did not get inlined/evaluated. The data is also added to snapshots to further improve regression testing too.

### CODE REVIEW

**Comment by gaearon** [comment]
Some snapshots are failing?

**Comment by trueadm** [comment]
@gaearon Had to update snapshots, done now!

File: scripts/__snapshots__/test-react.js.snap [hunk 1]
```diff
@@ -2,1903 +2,5811 @@
...
 
 exports[`Test React with JSX input, create-element output fb-www mocks fb-www 3 1`] = `
 ReactStatistics {
+  "evaluatedRootNodes": Array [],
   "inlinedComponents": 0,
   "optimizedTrees": 0,
 }
 `;
 
 exports[`Test React with JSX input, create-element output fb-www m
```
Comment (gaearon) [inline scripts/__snapshots__/test-react.js.snap]: Can we pass this through pretty printer before giving it to jest?

Comment (trueadm) [reply scripts/__snapshots__/test-react.js.snap]: I couldn't see how we could, as the snapshot applies its own formatting.

File: src/realm.js [hunk 2]
```diff
@@ -188,7 +188,7 @@ export class Realm {
       output: opts.reactOutput || "create-element",
       hoistableFunctions: new WeakMap(),
       hoistableReactElements: new WeakMap(),
-      reactElements: new Set(),
+      reactElements: new WeakSet(),
```
Comment (NTillmann) [inline src/realm.js]: What's this change about? Doesn't seem related.
